libs/api_common/device_helper.cpp

```diff
@@ -129,6 +129,10 @@ register_wait_async_ioctl_operation(_Inout_ async_ioctl_completion_t* async_ioct
 
     // Set the event on the thread-pool wait object.
     ebpf_assert(async_ioctl_completion->wait != nullptr);
+    if (async_ioctl_completion->wait == nullptr) {
```

Comment:
Do we need both the assert and the if check for `async_ioctl_completion->wait`.
I think if we are adding `if` condition then we don't need the assert.